### PR TITLE
Upgrade to govuk-frontend 5.2.0

### PR DESCRIPTION
**Pull request by peteryates**
- Upgrade to govuk-frontend version 5.2.0
- Upgrade form-builder and components to 5.2.0

~~Failing because of asset paths and I have no intention of learning Webpacker. Should we replace it with something more modern? cc @slawosz @javier-npq @pmanrubia~~

I have fixed the paths for now but agree we should upgrade.

### Header

| Before | After |
| ----- | ----- |
| Screenshot from 2024-02-23 10-26-34 |Screenshot from 2024-02-23 10-26-30 |
| Screenshot from 2024-02-23 11-29-18 | Screenshot from 2024-02-23 11-29-22 |

* fix weird alignment of menu
* lose the always-present menu dropdown in the top right - the menu will appear automatically when the screen size is small enough

### Footer

| Before | After |
| ----- | ----- |
| Screenshot from 2024-02-23 11-09-57 | Screenshot from 2024-02-23 16-50-34 |

* ~~'Support and guidance' messaging is missing for now, I'll add it back once [the corresponding branch](https://github.com/x-govuk/govuk-components/pull/511) in govuk-components has been merged~~

## Final tasks

* [x] carefully go through the GOV.UK frontend release notes to see if any things not covered by the gems have been missed
* [x] Add the support and guidance section back to the footer

### CODE REVIEW

**Comment by peteryates** [comment]
Please do, I'm unfamiliar with the JS in the app so didn't want to do it, but I'd be very happy with a switch.

File: spec/helpers/identity_account_helper_spec.rb [hunk 1]
```diff
@@ -32,4 +32,23 @@
       expect(link).to match("sign_out_uri=#{url_encoded}")
     end
   end
+
+  describe "#identity_link_uri" do
+    let(:fake_acocunt_link) do
```
Comment (ethax-ross) [inline spec/helpers/identity_account_helper_spec.rb:37]: typo

**Review by ethax-ross — APPROVED** [review]
👏 Looks good! just a typo spotted

File: spec/helpers/identity_account_helper_spec.rb (hunk 1 above)
Comment (peteryates) [reply spec/helpers/identity_account_helper_spec.rb:37]: And a rude one at that 🫣

Will fix right away!

**Pull request merged**